### PR TITLE
Server routes added for more functionality.

### PR DESCRIPTION
**Pull request by moods-72**
I added server routes to interact with the frontend and database. I used 'bcrypt' for the password hashing.  I have to work on the cookie parsing.

### CODE REVIEW

File: server/routes/login.js [hunk 1]
```diff
@@ -0,0 +1,19 @@
+const express = require("express");
+const router = express.Router();
+const bcrypt = require("bcrypt");
+const User = require("../models/User");
+
+router.get("/:email/:password", async (req, res, next) => {
```
Comment (dproc96) [inline server/routes/login.js:6]: These should be passed in the body of a POST request, not as params

File: server/routes/signup.js [hunk 2]
```diff
@@ -0,0 +1,32 @@
+const express = require("express");
+const router = express.Router();
+const bcrypt = require("bcrypt");
+const saltRounds = 10;
+const User = require("../models/User");
+
+router.post("/:email/:password", (req, res, next) => {
```
Comment (dproc96) [inline server/routes/signup.js]: Same as above, use the request body

File: server/routes/signup.js [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+  const { email, password } = req.params;
+  let epCheck = false;
+  if (email.includes("@") && password.length > 6 && (epCheck = true)) {
+    bcrypt.hash(password, saltRounds, (err, hash) => {
+      const addUser = async () => {
+        const user1 = new User({
```
Comment (dproc96) [inline server/routes/signup.js]: You can just call this user, user1 looks a little odd

File: server/routes/signup.js [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+          email: email,
+          password: hash,
+        });
+        await user1
```
Comment (dproc96) [inline server/routes/signup.js]: You don't need this await if you're using a .then() block, this also means it doesn't need to be in an async function

File: server/routes/signup.js [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+          .save()
+          .then(() => res.status(200).send({ msg: "Account created!." }))
+          .catch((err) => {
+            res.status(401).send({ msg: err });
+          });
+      };
+      addUser();
```
Comment (dproc96) [inline server/routes/signup.js]: It's weird to define the function right above and then call it here once, would be better to just keep it all in the same block

File: server/routes/signup.js [hunk 6]
```diff
@@ -0,0 +1,32 @@
+const express = require("express");
+const router = express.Router();
+const bcrypt = require("bcrypt");
+const saltRounds = 10;
+const User = require("../models/User");
+
+router.post("/:email/:password", (req, res, next) => {
+  const { email, password } = req.params;
+  let epCheck = false;
+  if (email.includes("@") && password.length > 6 && (epCheck = true)) {
```
Comment (dproc96) [inline server/routes/signup.js]: It's not good practice to put an assignment inside of a conditional block like this. You can use and else block to accomplish the same thing

File: server/routes/findUser.js [hunk 7]
```diff
@@ -0,0 +1,14 @@
+const express = require("express");
+const router = express.Router();
+const User = require("../models/User");
+
+router.get("/:email", async (req, res, next) => {
```
Comment (dproc96) [inline server/routes/findUser.js]: For getting a user it's better to use something immutable like id. You could also consolidate the find and delete routes into one userRouter to improve readability

**Review by dproc96 — CHANGES_REQUESTED** [review]
This is a really good start! I had a couple notes on routing conventions that should be addressed

**Comment by dproc96** [comment]
There are also two other parts of this ticket that need to be added, an authentication middleware and sending the token to the client

File: server/routes/findUser.js (hunk 7 above)
Comment (moods-72) [reply server/routes/findUser.js]: Thanks for the encouragement.  Updates completed.

**Comment by moods-72** [comment]
The authentication is giving me problems. I believe it's a CORS issue, and the steps I have tried are not working. My own ideas and internet searches have not brought me the solution(s) as of yet. It seems so simple, but it is perplexing me.

File: client/src/pages/Login.jsx [hunk 8]
```diff
@@ -21,8 +22,15 @@ function Login() {
       : setPasswordError("Password must be > 6 characters.");
     if (!emailError.length && !passwordError.length) {
       axios
-        .get(`http://localhost:3001/login/${email}/${password}`)
-        .then((data) => console.log(data.data[0]))
+        .post("http://localhost:3001/login/", {
```
Comment (dproc96) [inline client/src/pages/Login.jsx:25]: This is probably your CORS issue, try just doing "/login" or window.location.origin + "/login"

File: server/app.js [hunk 9]
```diff
@@ -31,8 +31,8 @@ app.use("/", indexRouter);
 app.use("/ping", pingRouter);
 app.use("/login", loginRouter);
 app.use("/signup", signupRouter);
-app.use("/find", findUserRouter);
-app.use("/delete", deleteUserRouter);
+app.use("/findDeleteUser", findDeleteUserRouter);
```
Comment (dproc96) [inline server/app.js:34]:
```suggestion
app.use("/user", userRouter);
```
This is more extendable as we will probably want more options in the future

File: server/routes/findDeleteUser.js [hunk 10]
```diff
@@ -0,0 +1,20 @@
+const express = require("express");
+const router = express.Router();
+const User = require("../models/User");
+
+router.get("/", async (req, res, next) => {
```
Comment (dproc96) [inline server/routes/findDeleteUser.js:5]: Best to have a /:id route and a /delete/:id route

File: server/routes/signup.js [hunk 11]
```diff
@@ -1,27 +1,22 @@
...
+  if (email.includes("@") && password.length > 6) {
+    epCheck = true;
+    bcrypt.hash(password, 10, (err, hash) => {
+      const user = new User({
+        email: email,
+        password: hash,
+      });
+      user
+        .save()
+        .then(() => res.status(200).send({ msg: "Account created!." }))
```
Comment (dproc96) [inline server/routes/signup.js:18]: You can send the user and the auth token in this response

**Comment by dproc96** [comment]
From what you've said this PR is no longer necessary because all of the changes were in the one merged to dev yesterday. Please reopen if I'm incorrect

**Pull request closed without merging**